### ISSUE TITLE
VDC list of storage profiles needs to be a TypeSet

### ISSUE THREAD

**Issue by dataclouder** [vmware/terraform-provider-vcd#404]
Using the same setup as for issue #403, a VDC plan will ask for a rebuild because of the storage profiles positions.

```hcl
resource "vcd_org_vdc" "vdc-datacloud" {
  name = "vdc-datacloud"
  org  = vcd_org.datacloud.name

  allocation_model  = "AllocationVApp"
  provider_vdc_name = "PVDC"
  network_pool_name = "PVDC-VXLAN-NP"
  network_quota     = 50

  compute_capacity {
    cpu {
      limit = 0
    }

    memory {
      limit = 0
    }
  }

  storage_profile {
    name    = "Development"
    enabled = true
    limit   = 0
    default = false
  }

  storage_profile {
    name    = "*"
    enabled = true
    limit   = 0
    default = true
  }

  enabled                  = true
  enable_thin_provisioning = true
  enable_fast_provisioning = true
  delete_force             = true
  delete_recursive         = true
}
```

When we run `terraform plan`, we get

```
# vcd_org_vdc.vdc-datacloud will be updated in-place
  ~ resource "vcd_org_vdc" "vdc-datacloud" {
        allocation_model         = "AllocationVApp"
        cpu_guaranteed           = 0.2
        cpu_speed                = 1000
        delete_force             = true
        delete_recursive         = true
        enable_fast_provisioning = true
        enable_thin_provisioning = true
        enable_vm_discovery      = false
        enabled                  = true
        id                       = "urn:vcloud:vdc:834d62cc-3cb6-4247-9391-1861462db370"
        memory_guaranteed        = 0.2
        metadata                 = {}
        name                     = "vdc-datacloud"
        network_pool_name        = "PVDC-VXLAN-NP"
        network_quota            = 50
        nic_quota                = 0
        org                      = "datacloud"
        provider_vdc_name        = "PVDC"
        vm_quota                 = 0

        compute_capacity {
            cpu {
                allocated = 0
                limit     = 0
                overhead  = 0
                reserved  = 0
                used      = 0
            }

            memory {
                allocated = 0
                limit     = 0
                overhead  = 0
                reserved  = 0
                used      = 0
            }
        }

      ~ storage_profile {
          ~ default = false -> true
            enabled = true
            limit   = 0
          ~ name    = "Development" -> "*"
        }
      ~ storage_profile {
          ~ default = true -> false
            enabled = true
            limit   = 0
          ~ name    = "*" -> "Development"
        }
    }
```

**Comment by lvirbalas** [comment]
Note, that changing to `TypeSet` would mean a single field change, which is now rendered like this:
```
      ~ storage_profile {
            default = false
            enabled = true
          ~ limit   = 0 -> 1000
            name    = "Development"
        }
```
Would look like a whole block change due to how Terraform `TypeSet` works.

**Comment by vbauzysvmware** [comment]
fixed and will be released in 3.1 version
